### PR TITLE
Add guidelines for modal dialogs

### PR DESCRIPTION
**Pull request by ntrogh**
Fixes https://github.com/microsoft/vscode-docs/issues/4425

### CODE REVIEW

**Comment by ntrogh** [comment]
@isidorn Can you take a look at this guidance for modals?

File: api/ux-guidelines/notifications.md [hunk 1]
```diff
@@ -69,6 +69,24 @@ When needing to display progress for an indeterminate timeframe (for example, se
...
+
+**✔️ Do**
+
+* Only use modal dialogs if you need immediate user interaction
+* Where appropriate, provide an action to avoid repeated user confirmation (*Always*/*Never* action)
+
+**❌ Don't**
+
+* Use modal dialogs to confirm multiple steps
+* Use modal dialogs for showing messages that don't require an action from the user
```
Comment (isidorn) [inline api/ux-guidelines/notifications.md:88]: We could add a Don't

"Show model dialogs for actions that user did not explicitly initiate"

File: api/ux-guidelines/notifications.md [hunk 2]
```diff
@@ -69,6 +69,24 @@ When needing to display progress for an indeterminate timeframe (for example, se
 
 *This example uses the progress notification to show the setup involved for a remote connection, while also providing a link to the output logs (**details**).*
 
+## Modal dialog
+
+When you need immediate user input for an action, you can opt to show a modal dialog. This UI element should be used with caution because a modal dialog blocks user interactions outside the dialog, until it's dismissed.
+
+![Modal dialog](images/examples/javascript-update-imports-dialog.png)
```
[the remark this answers is not in the document]
Comment (ntrogh) [reply api/ux-guidelines/notifications.md:76]: What about this one from Copilot. It has a confirmation, decline, and checkbox.

image

File: api/ux-guidelines/notifications.md (hunk 1 above)
Comment (ntrogh) [reply api/ux-guidelines/notifications.md:88]: Good one - will add

File: api/ux-guidelines/notifications.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (isidorn) [reply api/ux-guidelines/notifications.md:76]: This is good. We can also in guidelines have "consider using checkboxes for preserving user choice"

**Review by isidorn — APPROVED** [review]

**Pull request merged**